### PR TITLE
enh(conf/export) tags and severity export for Ressource Status

### PR DESCRIPTION
**Pull request by a-launois**
## Description

export tags and severity in host, hostTemplates, and serviceTemplates config files

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by kduret — DISMISSED** [review]

File: www/class/config-generate/abstract/host.class.php [hunk 1]
```diff
@@ -395,4 +396,45 @@ public function getString($host_id, $attr)
         }
         return null;
     }
+
+    /**
+     * @param array<string,mixed> $host
+     * @return array<string,mixed>
+     */
+    private function getHostCategoriesByHost(array $host): array
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:404]:
```suggestion
    private function getHostCategoriesByHost(array $host): ?array
```

File: www/class/config-generate/abstract/host.class.php [hunk 2]
```diff
@@ -395,4 +396,45 @@ public function getString($host_id, $attr)
         }
         return null;
     }
+
+    /**
+     * @param array<string,mixed> $host
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:401]:
```suggestion
     * @param array<string,mixed>|null $host
```

File: www/class/config-generate/abstract/host.class.php [hunk 3]
```diff
@@ -395,4 +396,45 @@ public function getString($host_id, $attr)
...
+                WHERE host_host_id = :host_id"
+            );
+            $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $stmt->execute();
+            $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);
+        } else {
+            $hostCategories = $host['hostCategories'];
+        }
+
+        return $hostCategories ?? null;
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:419]:
```suggestion
        return $hostCategories;
```

File: www/class/config-generate/abstract/host.class.php [hunk 4]
```diff
@@ -395,4 +396,45 @@ public function getString($host_id, $attr)
...
+            );
+            $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $stmt->execute();
+            $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);
+        } else {
+            $hostCategories = $host['hostCategories'];
+        }
+
+        return $hostCategories;
+    }
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:420]:
```suggestion
        }

        return $hostCategories ??  $host['hostCategories'];
    }
```

File: www/class/config-generate/abstract/host.class.php [hunk 5]
```diff
@@ -395,4 +396,43 @@ public function getString($host_id, $attr)
...
+                "SELECT hostcategories_hc_id
+                FROM hostcategories_relation
+                WHERE host_host_id = :host_id"
+            );
+            $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $stmt->execute();
+            $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);
+        }
+
+        return $hostCategories ??  $host['hostCategories'];
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:417]:
```suggestion

        if (isset($host['hostCategories'])) {
           return $host['hostCategories'];
        }
        $stmt = $this->backend_instance->db->prepare(
            "SELECT hostcategories_hc_id
            FROM hostcategories_relation
            WHERE host_host_id = :host_id"
        );
        $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
        $stmt->execute();
        $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);

        return $hostCategories ;
```

File: www/class/config-generate/abstract/host.class.php [hunk 6]
```diff
@@ -395,4 +396,44 @@ public function getString($host_id, $attr)
...
+        $stmt = $this->backend_instance->db->prepare(
+            "SELECT hostcategories_hc_id
+            FROM hostcategories_relation
+            WHERE host_host_id = :host_id"
+        );
+        $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+        $stmt->execute();
+        $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);
+
+        return $hostCategories ;
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:418]:
```suggestion
        return $hostCategories;
```

File: www/class/config-generate/abstract/host.class.php [hunk 7]
```diff
@@ -395,4 +396,44 @@ public function getString($host_id, $attr)
...
+        $stmt = $this->backend_instance->db->prepare(
+            "SELECT hostcategories_hc_id
+            FROM hostcategories_relation
+            WHERE host_host_id = :host_id"
+        );
+        $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+        $stmt->execute();
+        $hostCategories = $stmt->fetchAll(PDO::FETCH_COLUMN);
+
+        return $hostCategories;
```
Comment (a-launois) [inline www/class/config-generate/abstract/host.class.php:418]:
```suggestion
        return $stmt->fetchAll(PDO::FETCH_COLUMN);
```

File: www/class/config-generate/hostcategory.class.php [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+     * @param string $hostName
+     */
+    public function insertHostToCategoryMembers(int $hostCategoryId, int $hostId, string $hostName): void
+    {
+        if (! isset($this->hostCategories[$hostCategoryId])) {
+            $this->addHostCategoryToList($hostCategoryId);
+        }
+        if (
+            isset($this->hostCategories[$hostCategoryId])
+            && ! isset($this->hostCategories[$hostCategoryId]['members'][$hostId])
```
Comment (a-launois) [inline www/class/config-generate/hostcategory.class.php:98]:
```suggestion
        if (! isset($this->hostCategories[$hostCategoryId])) {
            $this->addHostCategoryToList($hostCategoryId);
        } elseif ( ! isset($this->hostCategories[$hostCategoryId]['members'][$hostId])
```

Comment (a-launois) [reply www/class/config-generate/hostcategory.class.php:98]: idem in servicecategory

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**